### PR TITLE
Add release stage to version

### PR DESCRIPTION
**Pull request by nayres**
🎟️ [Asana Task](https://app.asana.com/0/1203347061500809/1203374761617827)
🔍 [Preview Link](https://react-components-add-release-stage-to-version-hashicorp.vercel.app)

---

## Description

Add releaseStage to `mapVersionList` version label for non stable/ non latest versions

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/docs-page/server/loaders/remote-content.test.ts [hunk 1]
```diff
@@ -256,92 +258,48 @@ describe('RemoteContentLoader', () => {
...
-    expect(versionList[0].label.endsWith('(latest)')).toBe(true)
+  test('should label the latest version as "latest"', () => {
+    const latestVersion = versionList.find(
+      (version) => version.isLatest === true
+    )
+
+    expect(latestVersion?.label.endsWith('(latest)')).toBe(true)
   })
 
-  test('should sort by semver descending', () => {
```
Comment (thiskevinwang) [inline packages/docs-page/server/loaders/remote-content.test.ts:266]: 👍 (This sorting logic predates sorting at the API level. This is good to remove)

File: packages/docs-page/server/loaders/remote-content.ts [hunk 2]
```diff
@@ -79,22 +97,18 @@ const cachedFetchVersionMetadataList = moize(
...
-    .sort((a, b) =>
-      semver.compare(semver.coerce(b.version)!, semver.coerce(a.version)!)
-    )
-    .map((e) => {
-      const { isLatest, version, display } = e
-
-      const displayValue = display || version
+  const versions = list.map((versionOption) => {
+    const { isLatest, version, display, releaseStage } = versionOption
+    const displayValue = display || version
```
Comment (thiskevinwang) [inline packages/docs-page/server/loaders/remote-content.ts:102]: 💭 Given that `displayValue` is derived from `versionOption`, maybe you can lift this into `determineLabel(option)`, remove the second arg

File: packages/docs-page/server/loaders/remote-content.ts [hunk 3]
```diff
@@ -79,22 +97,18 @@ const cachedFetchVersionMetadataList = moize(
...
-        label: isLatest ? `${displayValue} (latest)` : displayValue,
-        isLatest: isLatest || false,
-        version,
-      }
-    })
+    return {
+      name: isLatest ? 'latest' : version,
+      label: determineLabel(versionOption, displayValue),
+      isLatest: isLatest || false,
+      releaseStage: releaseStage as ReleaseStage,
```
Comment (thiskevinwang) [inline packages/docs-page/server/loaders/remote-content.ts:108]: ❔ is this `as ReleaseStage` type-casting still necessary? It _looks like_ the type should be correct even with out the `as ...`.

**Review by thiskevinwang — COMMENTED** [review]
Looks pretty good! Left a few questions/comments

File: packages/docs-page/server/loaders/remote-content.ts (hunk 2 above)
Comment (nayres) [reply packages/docs-page/server/loaders/remote-content.ts:102]: Good catch! Was wondering what to do with displayValue, but didn't think of this somehow!

**Review by thiskevinwang — APPROVED** [review]
LGTM — nice work 👍

**Pull request merged**